### DISCUSSION TITLE
Enable smartphone vibrate on Android on fcm retrieval

### DISCUSSION THREAD

**Discussion by andyzukunft** [TobiasBuchholz/Plugin.Firebase#467] [General]
I noticed that the phone is not vibrating on fcm notification arrival (app is in background). The app should have the permission, the channel is configured to vibrate and I checked the app permission on the phone where "Sound and Vibration" is activated in Android.
The current implementation only creates a notification sound and of course only if the phone is set to "loud" however a lot of phones nowadays are set to "silent / vibration" which results in the app not giving any indication that a notification is waiting.

Do you have an idea what to add/fix to enable vibrations for Android? Do we have to do anything for iOS?

AndroidManifest.xml
```xml
<uses-permission android:name="android.permission.VIBRATE" />
```

Android MainActivity.cs
```csharp
Log.Debug($"Init FCM Channel");
var channelId = $"{PackageName}.general";

var channel = new NotificationChannel(channelId, "General", NotificationImportance.Default);
channel.EnableVibration(true);
channel.EnableLights(true);
channel.LockscreenVisibility = NotificationVisibility.Public;

var notificationManager = (NotificationManager)GetSystemService(NotificationService);
notificationManager.CreateNotificationChannel(channel);

FirebaseCloudMessagingImplementation.ChannelId = channelId;
FirebaseCloudMessagingImplementation.SmallIconRef = Resource.Drawable.ic_notification_small;
```

**Comment by kovenko** [comment]
Vibration.Vibrate();

**Reply by ideeyn** [reply]
yes it can work if it have data, wake the app first and runs a custom method that catches it.... but if the FCM is notification only? also kindly give more context to your code
